extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/MSQTaskQueryMaker.java

```diff
@@ -246,7 +246,8 @@ public QueryResponse<Object[]> runQuery(final DruidQuery druidQuery)
           targetDataSource.getDestinationName(),
           segmentGranularityObject,
           segmentSortOrder,
-          replaceTimeChunks
+          replaceTimeChunks,
+          null
```

Comment:
There should be a SegmentMorphFactorySupplier here which gets injected.

Reply:
That way you can add tests.

Reply:
Added